src/parser.rs

```diff
@@ -454,17 +496,29 @@ impl<'a, T: Iterator<Item = char>> Parser<'a, T> {
                                 for s in srcs.iter() {
                                     src.push_str(s);
                                 }
-
-                                self.tokens.push(Token::Section(
-                                    name,
-                                    inverted,
-                                    children,
-                                    self.opening_tag.clone(),
-                                    osection,
-                                    src,
-                                    tag,
-                                    self.closing_tag.clone(),
-                                ));
+                                self.tokens.push(if !name.is_empty() && name[0] == "-top-" {
```

Comment:
Maybe something like this instead:

```suggestion
                                self.tokens.push(if name.first() == Some("-top-") {
```